From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: mm: cma: report correct node id
Date: Thu, 19 Oct 2023 09:32:53 +0800

cma_declare_contiguous_nid()'s printk can show an invalid node ID:

cma: Reserved 32 MiB at 0x00000000fe000000 on node -1

Use early_pfn_to_nid() to get correct node id from base instead of the
default NUMA_NO_NODE in cma_declare_contiguous_nid(), so users will
instead see

cma: Reserved 32 MiB at 0x00000000fe000000 on node 0

Link: https://lkml.kernel.org/r/20231019013253.2792048-1-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Cc: Christoph Hellwig <hch@lst.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/cma.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/mm/cma.c~mm-cma-report-correct-node-id
+++ a/mm/cma.c
@@ -375,6 +375,9 @@ int __init cma_declare_contiguous_nid(ph
 	if (ret)
 		goto free_mem;
 
+	if (nid == NUMA_NO_NODE)
+		nid = early_pfn_to_nid(PHYS_PFN(base));
+
 	pr_info("Reserved %ld MiB at %pa on node %d\n", (unsigned long)size / SZ_1M,
 		&base, nid);
 	return 0;
_
